src/Planners/src/UnicycleTrajectoryPlanner.cpp

```diff
@@ -0,0 +1,841 @@
+                                             leftContactList))
+    {
+        log()->error("{} Error while getting the left contact list.", logPrefix);
+        return false;
+    };
+
+    if (!Planners::Utilities::getContactList(initTime,
+                                             dt,
+                                             rightFootInContact,
+                                             rightSteps,
+                                             m_pImpl->parameters.rightContactFrameIndex,
+                                             "right_foot",
+                                             rightContactList))
+    {
+        log()->error("{} Error while getting the right contact list.", logPrefix);
+        return false;
+    };
+
+    ContactListMap["left_foot"] = leftContactList;
+    ContactListMap["right_foot"] = rightContactList;
+    m_pImpl->output.ContactPhaseList.setLists(ContactListMap);
+    m_pImpl->output.steps.leftSteps = leftSteps;
+    m_pImpl->output.steps.rightSteps = rightSteps;
+
+    // get the DCM trajectory
+    auto convertToEigen
+        = [](const std::vector<iDynTree::Vector2>& inputVect) -> std::vector<Eigen::Vector2d> {
+        std::vector<Eigen::Vector2d> outputVect;
+        outputVect.reserve(inputVect.size());
+
+        for (const auto& v : inputVect)
+        {
+            outputVect.push_back(iDynTree::toEigen(v));
+        };
+
+        return outputVect;
+    };
+
+    std::vector<Eigen::Vector2d> dcmPosition, dcmVelocity;
+    dcmPosition = convertToEigen(dcmGenerator->getDCMPosition());
+    dcmVelocity = convertToEigen(dcmGenerator->getDCMVelocity());
+    m_pImpl->output.dcmTrajectory.dcmPosition = dcmPosition;
+    m_pImpl->output.dcmTrajectory.dcmVelocity = dcmVelocity;
+
+    // get the CoM planar trajectory
+    auto time = std::chrono::nanoseconds(static_cast<int>(initTime * 1e9));
+    Eigen::Vector4d state;
+    state.head(2) = m_pImpl->input.comInitialState.initialPlanarPosition;
+    state.tail(2) = m_pImpl->input.comInitialState.initialPlanarVelocity;
+    m_pImpl->comSystem.dynamics->setState({state.head(4)});
+    using namespace BipedalLocomotion::GenericContainer::literals;
+    auto stateDerivative = BipedalLocomotion::GenericContainer::make_named_tuple(
+        BipedalLocomotion::GenericContainer::named_param<"dx"_h, Eigen::VectorXd>());
+    Eigen::Vector4d controlInput;
+    std::vector<Eigen::Vector2d> comPlanarPosition, comPlanarVelocity, comPlanarAcceleration;
+
+    for (size_t i = 0; i < dcmPosition.size(); i++)
+    {
+        // populate CoM planar position
+        comPlanarPosition.push_back(state.head<2>());
+
+        // set the control input, u
+        controlInput << dcmPosition.at(i), dcmVelocity.at(i);
+        m_pImpl->comSystem.dynamics->setControlInput({controlInput});
+
+        // compute the state derivative xdot = Ax + Bu
+        m_pImpl->comSystem.dynamics->dynamics(time, stateDerivative);
```

Comment:
It is not clear to me why you manually compute the dynamics, could you explain it more in detail? Is it required to get the CoM planar velocity and acceleration?

Reply:
My idea was that the planner should provide also the CoM velocity and acceleration in case they are needed by any of the control architectures which will be used.